### PR TITLE
JSON API: Remove obsolete code to query invites directly from a JP site

### PR DESCRIPTION
**Pull request by nylen**
All invite data has been stored on WP.com for some time now.  This PR assumes this is true for multisite Jetpack installations as well, which appears to be the case because I can't find any Jetpack code that calls this endpoint.

If this is indeed the case, then we can simplify this endpoint pretty dramatically (short of a full removal of the endpoint from Jetpack, which would be another option).

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-list-invites-endpoint.php [hunk 1]
```diff
@@ -43,18 +43,17 @@ function callback( $path = '', $blog_id = 0 ) {
 			return $blog_id;
 		}
 
-		if ( ! is_multisite() ) {
-			return new WP_Error( 'forbidden', 'To query invites, site must be on a multisite installation.', 403 );
+		if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+			return new WP_Error( 'not_implemented', 'Querying WP.com endpoint data from a Jetpack site is not implemented.', 400 );
```
Comment (nylen) [inline json-endpoints/class.wpcom-json-api-list-invites-endpoint.php]: This should say "invite" instead of "endpoint".

**Comment by ebinnion** [comment]
@nylen Apologies for the slow reply. I've added a task to review this tomorrow.

**Comment by nylen** [comment]
No worries.  After thinking about this some more, I think we should remove this endpoint from Jetpack entirely and moving it to WPCOM-only.  If I am understanding the situation correctly, it hasn't been able to do anything for a good while.

**Comment by ebinnion** [comment]
> After thinking about this some more, I think we should remove this endpoint from Jetpack entirely and moving it to WPCOM-only

I agree, especially since we're having to add `?force_wpcom=true`

**Comment by nylen** [comment]
We could also remove this one:  https://github.com/Automattic/jetpack/blob/master/json-endpoints/class.wpcom-json-api-update-invites-endpoint.php

However, there's not as much of a reason to do it, because as far as I can tell, we don't need to make any changes to that endpoint to support new invite features.

**Pull request merged**